thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/components/WoWStdRuleDetector.java

```
@@ -0,0 +1,275 @@
+    df.mapInPlace(BooleanSeries.ALL_TRUE, COL_ANOMALY, COL_PATTERN, COL_DIFF_VIOLATION);
+
+    // Anomalies
+    List<MergedAnomalyResultDTO> anomalyResults = DetectionUtils.makeAnomalies(slice, df, COL_ANOMALY,
+        window.getEndMillis(),
+        DetectionUtils.getMonitoringGranularityPeriod(timeGranularity.toAggregationGranularityString(),
+            datasetConfig), datasetConfig);
+    dfBase = dfBase.joinRight(df.retainSeries(COL_TIME, COL_CURR), COL_TIME);
+    DetectionResult dr1 = DetectionResult.from(anomalyResults, TimeSeries.fromDataFrame(dfBase));
+    return DetectionResult.from(anomalyResults, TimeSeries.fromDataFrame(dfBase));
+  }
+
+  private DataFrame computePredictionInterval(DataFrame inputDF, long windowStartTime, String timezone) {
+
+    DataFrame resultDF = new DataFrame();
+    //filter the data inside window for current values.
+    DataFrame forecastDF = inputDF.filter(new Series.LongConditional() {
+      @Override
+      public boolean apply(long... values) {
+        return values[0] >= windowStartTime;
+      }
+    }, COL_TIME).dropNull();
+
+    int size = forecastDF.size();
+    double[] baselineArray = new double[size];
+    double[] upperBoundArray = new double[size];
+    double[] lowerBoundArray = new double[size];
+    long[] resultTimeArray = new long[size];
+    double[] errorArray = new double[size];
+    double[] std = new double[size];
+    double[] mean = new double[size];
+
+    //get the trainingDF for each week, which is the number of lookback to 1 week before the each week predict start time
+    for (int k = 0; k < size; k++) {
+      DataFrame trainingDF;
+      trainingDF = getLookbackDF(inputDF, forecastDF.getLong(COL_TIME, k));
+      //the get historical WoW mean and std.
+      std[k]= trainingDF.getDoubles(COL_CHANGE).std().value();
+      mean[k] = trainingDF.getDoubles(COL_CHANGE).mean().value();
+
+      // We need at least 4 weeks of data
+      if (trainingDF.size() < 4) {
```

When will we have less than 4 weeks? Should validate the configuration in ahead instead of silently drop here.

---

I added spec validation part. Thanks!